### ISSUE TITLE
Price model category relation key

### ISSUE THREAD

**Issue by WILLEMSCHOLTEN** [OFFLINE-GmbH/oc-mall-plugin#625]
When running the following query to get products that are 'on sale' i found out that the relation key on the price model is missing for the category relation.

```
$result = \OFFLINE\Mall\Models\Product::with(['translations', 'image_sets.images', 'customer_group_prices', 'additional_prices'])
            ->whereHas('additional_prices', function ($q) {
                $q->whereHas('category', function ($q1) {
                    $q1->where('code', 'old_price');
                });
            })->limit(4)->get();
```

Current code @ plugins/offline/mall/models/Price.php line 28

```
public $belongsTo = [
        'category' => [PriceCategory::class],
        'currency' => [Currency::class],
    ];
```
Fix
```
public $belongsTo = [
        'category' => [PriceCategory::class, 'key' => 'price_category_id'],
        'currency' => [Currency::class],
    ];
```

**Comment by mjauvin** [comment]
Please, submit a PR to fix this.

**Comment by tobias-kuendig** [comment]
This was fixed in #626
